enterprise/server/test/integration/remote_bazel/remote_bazel_test.go

```diff
@@ -0,0 +1,208 @@
+	rootDir := "/root/workspace/remote-bazel-integration-test"
+	err := os.Setenv("HOME", rootDir)
+	require.NoError(t, err)
+
+	err = os.MkdirAll(rootDir, 0755)
+	require.NoError(t, err)
+
+	if _, err := os.Stat(fmt.Sprintf("%s/bazel-gazelle", rootDir)); os.IsNotExist(err) {
+		output := testshell.Run(t, rootDir, "git clone https://github.com/bazelbuild/bazel-gazelle --filter=blob:none --depth=1")
+		require.NotContains(t, output, "fatal")
+	}
+
+	err = os.Chdir(fmt.Sprintf("%s/bazel-gazelle", rootDir))
+	require.NoError(t, err)
+
+	// Run a server and executor locally to run remote bazel against
+	env, bbServer, _ := runLocalServerAndExecutor(t, "")
+	ctx := env.WithUserID(context.Background(), env.UserID1)
+	reqCtx := &ctxpb.RequestContext{
+		UserId:  &uidpb.UserId{Id: env.UserID1},
+		GroupId: env.GroupID1,
+	}
+
+	// Get an API key to authenticate the remote bazel request
+	bbClient := env.GetBuildBuddyServiceClient()
+	apiRsp, err := bbClient.CreateApiKey(ctx, &akpb.CreateApiKeyRequest{
+		RequestContext: reqCtx,
+		GroupId:        env.GroupID1,
+		Capability: []akpb.ApiKey_Capability{
+			akpb.ApiKey_CAS_WRITE_CAPABILITY,
+			akpb.ApiKey_CACHE_WRITE_CAPABILITY,
+			akpb.ApiKey_ORG_ADMIN_CAPABILITY,
+		},
+	})
+	require.NoError(t, err)
+	apiKey := apiRsp.ApiKey.Value
+
+	exitCode, err := remotebazel.HandleRemoteBazel([]string{
+		fmt.Sprintf("--remote_runner=%s", bbServer.GRPCAddress()),
+		// Have the ci runner use the "none" isolation type because it's simpler
+		// to setup than a firecracker runner
+		"--runner_exec_properties=workload-isolation-type=none",
+		"--runner_exec_properties=container-image=",
+		// Give more memory to the CI runner because it needs to clone the gazelle repo
+		"--runner_exec_properties=EstimatedComputeUnits=3",
+		"help",
+		fmt.Sprintf("--remote_header=x-buildbuddy-api-key=%s", apiKey)})
+	require.NoError(t, err)
+	require.Equal(t, 0, exitCode)
+
+	// Check the invocation logs to ensure the "bazel help" command successfully run
+	searchRsp, err := bbClient.SearchInvocation(ctx, &inpb.SearchInvocationRequest{
+		RequestContext: reqCtx,
+		Query:          &inpb.InvocationQuery{GroupId: env.GroupID1},
+	})
+	require.NoError(t, err)
+	require.Equal(t, 1, len(searchRsp.GetInvocation()))
+	invocationID := searchRsp.Invocation[0].InvocationId
+
+	logResp, err := bbClient.GetEventLogChunk(ctx, &elpb.GetEventLogChunkRequest{
+		InvocationId: invocationID,
+		MinLines:     math.MaxInt32,
+	})
+	require.NoError(t, err)
+	require.Contains(t, string(logResp.GetBuffer()), "Usage: bazel <command> <options>")
+}
+
+func TestWithPrivateRepo(t *testing.T) {
+	repoName := "private-test-repo"
+	username := "maggie-lou"
+	// This token grants read-only access to a private dummy repo.
+	// Github will invalidate tokens that are committed, so format it so that Github
+	// won't catch it.
+	personalAccessToken := fmt.Sprintf("github_pat_%s", "11ADKOFLI06Ob3cnrygcQa_gwLj0orOD3EILEu2k9pJ9EUkmXW6uM0k47k0uPOWInCDPHNC444JmQ8oN62")
```

Comment:
I think we should avoid checking in access tokens, even though it only grants access to an empty repo - could you use the gh app token env var that is already passed to the workflow, so that this authenticates as the app installation rather than a person?

Reply:
Could you explain more what you're describing? This is a /buildbuddy repo test, so are you describing the /buildbuddy app token? 

Also as a somewhat related note - I was playing around with accessing BB secrets from within a go test, and the only way I could get it to work was if I added `--test_env=SECRET_NAME` to the bazel command in buildbuddy.yaml. This passes the secret to all the tests, which may not be ideal (may be okay in this case, but could get messy if a lot of tests need to reference different secrets)

Reply:
Refactored to pull PAT from a secret